docs/API-Reference/bpmn-js/importXML.md

```diff
@@ -10,3 +10,62 @@ grand_parent: "API Reference"
 ---
 
 # importXML
+
+`importXML` API parses and renders a BPMN 2.0 diagram.
+
+## Usage
+
+importXML(xml: _[String](#)_, bpmnDiagram: _[ModdleElement](#)_ or _[String](#)_): Promise<_[ImportXMLResult](#importxmlresult)_, _[ImportXMLError](#importxmlerror)_>
+
+* `xml`: The BPMN 2.0 XML.
+* `bpmnDiagram`: The BPMN diagram or the id of the diagram to render _(if not provided, the first one will be rendered)_.
+
+### Code Example
+
+```javascript
+const modeler = new BpmnJS();
+
+try {
+
+  const result = await modeler.importXML(xml);
+
+  const { warnings } = result;
+
+  console.log('Import warnings: ', warnings);
+} catch (err) {
+
+  console.log('Error happened: ', err.message, err.warnings);
+}
+```
+
+### Life-Cycle Events
+
+During the import, these [events](/events) will be fired:
+
+* [import.parse.start](/events/bpmn-events/import-parse-start): About to read model from xml
+* [import.parse.complete](/events/bpmn-events/import-parse-complete): Model is read, either successfully or with errors.
+* [import.render.start](/events/bpmn-events/import-render-start): Graphical import is starting
+* [import.render.complete](/events/bpmn-events/import-render-complete): Graphical import is completed
+* [import.done](/events/bpmn-events/import-done): Everything is done.
+
+## Using with Callback Function
+
+With the [v7.0.0 release](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#700) public bpmn-js APIs are made awaitable and passing callbacks to APIs will be deprecated in a future release. See [here](https://bpmn.io/l/moving-to-promises.html) for more info.
+
+# ImportXMLResult
```

Comment:
Are we introducing these types with the docs or are they defined inside our codebase?

Reply:
I gave this topic some thought. We do the same in the documentation in our codebase to define certain types. It's important to state what the Promise resolves exactly so we should include types IMO. We don't claim we support Typescript anywhere and Typescript is yet-another-3rd-party-Javascript-related-tool which is pretty popular but it's still a 3rd party tool. I don't think we should go that far to shape the documented content based on a 3rd party tool.

One additional thing I like about this approach is that the users can click on the types and go to the type documentation via permalinks. So it's pretty useful.

What would be the alternative you propose if you want to chance our approach here?

Reply:
We use types in the code for things that like `djs.model.Base`, `ModdleElement` and `Event`. These things are actual prototypes, not one-off things. So I'd like us to find another solution for documenting this.

Reply:
Took care of this one 👍